game-core/src/main/java/games/strategy/triplea/delegate/move/validation/MoveValidator.java

```diff
@@ -278,13 +279,13 @@ private MoveValidationResult validateCanal(
    * @param units (Can be null. If null we will assume all units would be stopped by the canal.)
    */
   public String validateCanal(
-      final Route route, final Collection<Unit> units, final GamePlayer player) {
+      final Route route, @Nullable final Collection<Unit> units, final GamePlayer player) {
     return validateCanal(route, units, new HashMap<>(), player);
   }
 
   String validateCanal(
       final Route route,
-      final Collection<Unit> units,
+      @Nullable final Collection<Unit> units,
```

Comment:
Just noticed, when is the `units` value null? Generally an empty collection is preferred rather than a `null`.

Reply:
I don't know.  But one of the callers to this method has it flagged as `@Nullable` so I flagged this one as well since the caller can pass a null value.

Reply:
Fair enough, it is better safe than sorry.  :+1: